streams/src/main/java/org/apache/kafka/streams/processor/internals/ChangelogReader.java

```diff
@@ -21,29 +21,35 @@
 
 import java.util.Collection;
 import java.util.Map;
+import java.util.Set;
 
 /**
  * Performs bulk read operations from a set of partitions. Used to
- * restore  {@link org.apache.kafka.streams.processor.StateStore}s from their
- * changelogs
+ * restore a state store from {@link TopicPartition} changelog.
  */
 public interface ChangelogReader {
     /**
      * Register a state store and it's partition for later restoration.
-     * @param restorer the state restorer to register
+     *
+     * @param partition the changelog topic partition to restore
+     * @param stateManager the state manager used for restoring (one per task)
      */
-    void register(final StateRestorer restorer);
+    void register(final TopicPartition partition, final ProcessorStateManager stateManager);
```

Comment:
As mentioned in the description, we need to pass in the state manager as well since changelog reader is per-thread and hence could take on multiple tasks / state managers.